--- a/drivers/pmdomain/sunxi/Kconfig	2025-09-11 17:23:23.000000000 +0200
+++ b/drivers/pmdomain/sunxi/Kconfig	2026-04-04 02:25:08.520578756 +0200
@@ -18,3 +18,12 @@
 	  Say y to enable the Allwinner H6/H616 PRCM power domain driver.
 	  This is required to enable the Mali GPU in the H616 SoC, it is
 	  optional for the H6.
+
+config SUN50I_H713_PPU
+	bool "Allwinner H713 PPU power domain driver"
+	depends on ARCH_SUNXI || COMPILE_TEST
+	depends on PM
+	help
+	  Power domain driver for the Allwinner H713 SoC.
+	  Manages 5 power domains (CPUS, SYS, GPU, VE, DE)
+	  embedded in the R_CCU register space.
--- a/drivers/pmdomain/sunxi/Makefile	2025-09-11 17:23:23.000000000 +0200
+++ b/drivers/pmdomain/sunxi/Makefile	2026-04-04 02:25:08.544578850 +0200
@@ -1,3 +1,4 @@
 # SPDX-License-Identifier: GPL-2.0-only
 obj-$(CONFIG_SUN20I_PPU)		+= sun20i-ppu.o
 obj-$(CONFIG_SUN50I_H6_PRCM_PPU)	+= sun50i-h6-prcm-ppu.o
+obj-$(CONFIG_SUN50I_H713_PPU)	+= sun50i-h713-ppu.o
